### DISCUSSION TITLE
How to determine maximum send amount?

### DISCUSSION THREAD

**Discussion by noahNewton** [trustwallet/wallet-core#2145] [Q&A]
I'm trying to transfer the entire balance of ETH in our test wallet and I'm attempting to do this by subtracting the gas price from the total balance before sending and then sending that (balance - gasPrice) value in the "amount" param, but then I am faced with an error message that says: "insufficient funds or gas * price + value", what I am doing is subtracting the maxFeePerGas value from the amount value, so the actual params being sent would be {amount: (totalBalance - maxFeePerGas), maxFeePerGas: maxFeePerGas, ...restOfParams}, is there somewhere where I can find how your code is doing the (gas * price) + value calculation to see where maybe my gas calculation is off, though it is very close to the values on etherscan as I am using the latest block to derive the gas params, unsure what I'm doing wrong precisely.

Maybe to word this differently, how does trustwallet accomplish sending the entire balance in a transfer factoring in gas?

**Comment by hewigovens** [comment]
You need to estimate the gas limit first: `fee = (baseFee + priorityFee) * gasLimit`, Also it's tricky to max amount under EIP1559, need to make `baseFee = maxFee`
